src/Init/Data/BitVec/Lemmas.lean

```diff
@@ -781,6 +781,32 @@ protected theorem extractLsb_ofNat (x n : Nat) (hi lo : Nat) :
     (extractLsb' start len x).getLsbD i = (i < len && x.getLsbD (start+i)) := by
   simp [getLsbD, Nat.lt_succ]
 
+@[simp] theorem getMsbD_extractLsb'_eq_getLsbD {start len : Nat} {x : BitVec w} {i : Nat} :
+    (extractLsb' start len x).getMsbD i =
+    (decide (i < len) && x.getLsbD (start + (len - 1 - i))) := by
+  rcases len with _|len'
```

Comment:
```suggestion
  rcases len with rfl | len'
```
I prefer to match with `rfl` to make it clear that we are substituting the value `len = 0`.

Reply:
Since this is a purely auxiliary lemma to `getMsbD_extractLsb'`, it maybe worth refactoring the whole thing as a `where` clause:

```lean
@[simp] theorem getMsbD_extractLsb' {start len : Nat} {x : BitVec w} {i : Nat} :
    (extractLsb' start len x).getMsbD i = by <proof>
  where
     theorem getMsbD_extractLsb'_eq_getLsbD {start len : Nat} {x : BitVec w} {i : Nat} :
    (extractLsb' start len x).getMsbD i =
    (decide (i < len) && x.getLsbD (start + (len - 1 - i))) := by <proof>
```

Also, drop the `@[simp]` on `getMsbD_extractLsb'_eq_getLsbD`, since it's auxiliary!